### PR TITLE
docs: Update Triton Docs to New Theme

### PR DESCRIPTION
**Pull request by nv-kmcgill53**
#### What does the PR do?

https://github.com/triton-inference-server/server/pull/7807

#### Checklist
- [X] PR title reflects the change and is of format `<commit_type>: <Title>`
- [X] Changes are described in the pull request.
- [X] Related issues are referenced.
- [X] Populated [github labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) field
- [X] Added [test plan](#test-plan) and verified test passes.
- [X] Verified that the PR passes existing CI.
- [X] Verified copyright is correct on all changed files.
- [X] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- [X] All template sections are filled out.
- [ ] Optional: Additional screenshots for behavior/output changes with before/after.

#### Commit Type:
Check the [conventional commit type](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type)
box here and add the label to the github PR.
- [ ] build
- [ ] ci
- [X] docs
- [ ] feat
- [ ] fix
- [ ] perf
- [ ] refactor
- [ ] revert
- [ ] style
- [ ] test

#### Related PRs:

https://github.com/triton-inference-server/server/pull/7807

#### Test plan:

N/A

- CI Pipeline ID:

N/A

#### Caveats:

N/A

#### Background

N/A
#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)
N/A

### CODE REVIEW

File: docs/backend_guide/vllm.rst
Comment (mc-nv) [inline docs/backend_guide/vllm.rst:1]: Suppose choice of `rst` format is dictated to be more flexible on context path resolution, is it correct understanding?

**Review by statiraju — DISMISSED** [review]

**Comment by yinggeh** [comment]
Is there a pipeline for this change?

File: docs/backend_guide/vllm.rst
Comment (nv-kmcgill53) [reply docs/backend_guide/vllm.rst:1]: @statiraju to comment

Comment (statiraju) [reply docs/backend_guide/vllm.rst:1]: rst file was really needed to get the multiple levels on the left pane of the rendering to show multiple hierarchies.  spinx doesnt work well with md format for this.

**Review by statiraju — APPROVED** [review]

**Review by rmccorm4 — APPROVED** [review]
Approving to be in sync with what's already live on the docs for the release

**Pull request merged**